30413/Bogdan Emilian Sandu/Lab04/Mandatory/main.c

```diff
@@ -0,0 +1,92 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+typedef struct node
+{
+    int val;
+    struct node *right, *left;
+}NodeT;
+
+NodeT *createBinTree()
+{
+    int d;
+    scanf("%d",&d);
+    if(d==-1) return NULL;
+    NodeT *p;
+    p=(NodeT*)malloc(sizeof(NodeT));
+    if(p==NULL)
+        exit(1);
+    p->val=d;
+    p->left=createBinTree();
+    p->right=createBinTree();
+}
+int maxim(int a, int b)
+{
+    if(a>b) return a;
+    else return b;
+}
+
+void preorder(NodeT *p)
+{
+    if(p!=NULL)
+    {
+        printf("%d ",p->val);
+        preorder(p->left);
+        preorder(p->right);
+    }
+}
+void inorder(NodeT *p)
+{
+    if(p!=NULL)
+    {
+        inorder(p->left);
+        printf("%d ",p->val);
+        inorder(p->right);
+    }
+}
+void postorder(NodeT *p)
+{
+    if(p!=NULL)
+    {
+        postorder(p->left);
+        postorder(p->right);
+        printf("%d ",p->val);
+    }
+}
+void interChange(NodeT *p)
+{
+    NodeT *aux;
+    if(p==NULL) return;
+    aux=p->left;
+    p->left=p->right;
+    p->right=aux;
+}
```

Comment:
I am not sure if I understood right what was asked. Do I need to interchange only the 2 subtrees of the node between them or do I need to interchange all the subtress recursively from the node down.
In the second case I just need to add 2 more lines at the end of the function:
interChange(p->left);
interChange(p->right);

Reply:
Only interchange the 2 subtrees.